### ISSUE TITLE
Logs page keeps jumping to the bottom and loading domains - Chromium-based browser [Bug]

### ISSUE THREAD

**Issue by kevjaro** [hjk789/NXEnhanced#8]
I'm having an issue where the logs page keeps loading visited domains. The loading of visited domains usually only happens when you scroll down to the bottom of that page manually.  The script doesn't stop loading domains and it keeps dragging me further to the bottom of the page preventing me from using the Hide, Allow and Deny buttons of NXEnhancer.

It's as if the script is constantly trying to scroll down to the bottom of the page.

I'm having this issue on the Chromium-based Brave Browser on NXE 2.0 but I think I also encountered that issue with a previous version of NXE some time ago. No issue there on Firefox.

The issue stops when deactivating /* */ the following:

```javascript
// Prevent infinite scroll from being interrupted due to almost all queries being hidden

if (window.innerWidth == document.body.clientWidth) // If there is no vertical scrollbar, then surely the body height is insufficient to trigger the infinite scroll
{
	const dummyTallEll = document.createElement("div") // Create a temporary element to fill the empty space
	dummyTallEll.style.cssText = "height: " + (window.innerHeight - 300) + "px;"	// A static value is insufficient for big resolutions. This makes it relative to the window size
	document.querySelector(".Page .container .list-group").appendChild(dummyTallEll)
	scrollTo(0, document.body.clientHeight)
	dummyTallEll.remove()
}
```

Great job improving the usability of NextDNS btw! Thanks!

**Comment by hjk789** [comment]
Well, it does auto scroll, but it should only do it when 6 or less items are visible, when you're close to the end of the page and when there's no scroll bar in the page. You say it's always auto scrolling? You're using it on PC, right? What's your screen resolution?

**Comment by kevjaro** [comment]
Yep, using it on PC and there are always more than 6 items/ domains.

Resolution is 1920x1080.

**Comment by hjk789** [comment]
Strange. If you press the Home key or drag the scroll bar up to the top, does it still keep scrolling down or does it stop?   
Your browser window is maximized or resized?   
When you commented out the auto scroll code, did you commented out only the `if` content or the entire `if-else`?   
I tested on Chrome but it works fine, maybe it's specific to Brave Browser? I'll have to install it to test.

**Comment by kevjaro** [comment]
I had tested those already. After pressing the home key, it just jumps back to the bottom as further items load. Scrolling doesn't get me far either.

It's also the same wether the browser is maximised or resized.

I tried commenting out both `if` and `if - else`. The issue is solved if I comment out either.


It seems that I have found the issue though: I've tried running the script without any other extensions enabled and that solved it. I've got an [extension](https://github.com/iaarchiver/MinimalScrollbar) installed that hides the scrollbar when not scrolling or hovering over it with the cursor. NXE probably couldn't find it.

Not sure if there's a way to make it work (if scrollbar not found, ignore?) or if it's even worth the effort since I don't know how many people actually use scrollbar modifiers. I suppose it's just easier for me to disable that part of your code.

Thanks a lot for your help anyway! And for NXE!

**Comment by kevjaro** [comment]
Found a possible [solution](https://tylercipriani.com/blog/2014/07/12/crossbrowser-javascript-scrollbar-detection/):

```javascript
var hasScrollbar = window.innerWidth > document.documentElement.clientWidth
if (hasScrollbar){
                         //Prevent infinite scroll code
}
```

Maybe you can review this and see if it's worth adding?

**Comment by hjk789** [comment]
Oh, so that was it! I took a look at this extension and noticed that it disables the browser's native scrollbars and then simulates them using HTML and JavaScript. I tested here disabling the scroll bar, and doing this makes it behave the exact way you described. NXE always auto scrolls until it finds the scrollbars, but because they are disabled, it never finds it and keeps scrolling infinitely. 

Your code suggestion would have the same result, it's the same check but with an inverted logic.

Let's try this: in the `if` line, replace 
```javascript
if (window.innerWidth == document.body.clientWidth)
``` 
with:
```javascript
if (window.innerWidth == document.body.clientWidth && (document.body.clientHeight / window.innerHeight * 100) < 90)
```
Does it work now?
